src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -265,12 +265,6 @@ internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diag
 
             foreach (var diag in supportedDiagnostics)
             {
-                if (HasNotConfigurableTag(diag.CustomTags))
```

Comment:
we need to still see if there was a ruleset entry/compilation option for off-by default not configurable diagnostics. Try the below scenario for code you proposed:

1. Analyzer had diag1 which is configurable and diag2 which is not configurable. Both these diags are off by default.
2. User has a ruleset entry to turn diag2 on by having Action = Default. 

Your proposed change will deem this analyzer as suppressed, but actually it is not.